db_sql.go

```diff
@@ -648,28 +649,97 @@ func (x *sqlSessionDB) Read(sessionkey string) (*Token, error) {
 			return nil, err
 		}
 	} else {
-		if err := token.Permit.Deserialize(epermit); err != nil {
-			return nil, err
+		t, err2, done := x.populateToken(token, epermit, userId, oauthID)
```

Comment:
The current implementation of `populateToken` will always return true for `done`. I think you should rather check the `err`, like the previous code did.